kratos/utilities/pointer_communicator.h

```diff
@@ -78,23 +78,39 @@ class ResultsProxy
     /// Destructor.
     virtual ~ResultsProxy() {}
 
-    /**this function returns the effect of "user_function(gp)" both if the gp is locally owned
+    /**this function returns the effect of "user_function(rGlobalPointer)" both if the rGlobalPointer is locally owned
      * and if it is remotely owned
      */
-    TSendType Get(GlobalPointer<TPointerDataType>& gp)
+    TSendType Get(GlobalPointer<TPointerDataType>& rGlobalPointer)
     {
-        if(gp.GetRank() == mCurrentRank)
-            return mUserFunctor(gp);
+        if(rGlobalPointer.GetRank() == mCurrentRank)
+            return mUserFunctor(rGlobalPointer);
         else
-            return mNonLocalData[gp];
+            return mNonLocalData[rGlobalPointer];
```

Comment:
actually jere we should use the find, and have a debug check in the case it is not found

Reply:
Should we check `find` result to be valid always or only in debug?

Reply:
I put a KRATOS_DEBUG ward to check. This way in release the compiler should be able to generate exactly the same code as before (but const).

Considering all this we probably should also add a find/end (maybe changing the `end` for `invalid`, because if all are local values it is counter-intuitive) functions, as most of the time when I do `if(pc.has(x) then auto foo = pc.get(x)` could be rewrote as `auto foo = pc.find(x) if(foo != pc.end) ...` which is one less access.

But that can wait to other PR